src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.Tests.cs

```diff
@@ -79,6 +79,28 @@ public void Constructor_NoEntryInsertion_WritesNothing()
             Assert.Equal(0, archiveStream.Length);
         }
 
+        [Fact]
+        public void Write_To_UnseekableStream()
+        {
+            MemoryStream stream = new MemoryStream();
+            using WrappedStream wrapped = new WrappedStream(stream, canRead: true, canWrite: true, canSeek: false);
+
+            using (TarWriter writer = new TarWriter(wrapped, leaveOpen: true))
+            {
+                PaxTarEntry paxEntry = new PaxTarEntry(TarEntryType.RegularFile, "file.txt");
+                writer.WriteEntry(paxEntry);
+            } // The final records should get written, and the length should not be set because position cannot be read
+
+            stream.Seek(0, SeekOrigin.Begin); // Rewind the base stream (wrapped cannot be rewinded)
+
+            using TarReader reader = new TarReader(wrapped);
+            TarEntry entry = reader.GetNextEntry();
+            Assert.NotNull(entry);
```

Comment:
up to you, but I think many would not bother asserting entry is null, if they are going to get a NRE two lines later from `Assert.Equal(TarEntryType.RegularFile, entry.EntryType);.` maybe it's a style thing.

Reply:
I particularly like that VS Code lets you click on the filepath with the failure and it will take you directly to the assert that is failing. But in the end I think there's not much difference, I still need to look where that null object came from. So I think I can remove it as you suggest.

```
System.Formats.Tar.Tests.TarWriter_Tests.Write_To_UnseekableStream [FAIL]
        Assert.Null() Failure
        Stack Trace:
          /home/carlos/runtime/src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.Tests.cs(101,0): at System.Formats.Tar.Tests.TarWriter_Tests.Write_To_UnseekableStream()
             at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
          /home/carlos/runtime/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs(64,0): at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

vs

```
System.Formats.Tar.Tests.TarWriter_Tests.Write_To_UnseekableStream [FAIL]
        System.NullReferenceException : Object reference not set to an instance of an object.
        Stack Trace:
          /home/carlos/runtime/src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.Tests.cs(102,0): at System.Formats.Tar.Tests.TarWriter_Tests.Write_To_UnseekableStream()
             at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
          /home/carlos/runtime/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs(64,0): at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```